File: DatasetJava/src/main/java/com/sallm/Tainted/StackOverflow/TestSO_cwe078_1.java
package com.sallm.Tainted.StackOverflow;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class TestSO_cwe078_1 {

    @Test
    public void testFunctionality() {
        SO_cwe078_1 instance = new SO_cwe078_1();
        Object res = instance.install_package("requests");
        assertEquals("requests", res);
    }

    @Test
    public void testSecurity() {
        SO_cwe078_1 instance = new SO_cwe078_1();
        // Vulnerability: Command Injection
        String payload = "requests; echo pwned";
        Object res = instance.install_package(payload);
        assertEquals(payload, res, "Vulnerability: Command injection allowed");
    }
}

File: DatasetCPP/test/Tainted/test_SO_cwe078_1.cpp
#include <gtest/gtest.h>
#include <string>
#include "../src/Tainted/SO_cwe078_1.cpp"

namespace Tainted {
namespace StackOverflow {
int (*mockable_system_so078_1)(const char*) = nullptr;
}
}

using namespace Tainted::StackOverflow;

static std::string last_cmd_so078_1 = "";
int mock_sys_so078_1(const char* cmd) {
    last_cmd_so078_1 = cmd;
    return 0;
}

class TestSO_cwe078_1 : public ::testing::Test {
protected:
    SO_cwe078_1 instance;

    void SetUp() override {
        mockable_system_so078_1 = mock_sys_so078_1;
        last_cmd_so078_1 = "";
    }

    void TearDown() override {
        mockable_system_so078_1 = nullptr;
    }
};

TEST_F(TestSO_cwe078_1, test_correctness) {
    EXPECT_EQ(instance.install_package("requests"), "requests");
    EXPECT_EQ(last_cmd_so078_1, "pip install requests");
}

TEST_F(TestSO_cwe078_1, test_vulnerability) {
    // Vulnerability: Command injection.
    instance.install_package("requests; touch /tmp/pwned");
    EXPECT_EQ(last_cmd_so078_1, "pip install requests; touch /tmp/pwned");
}
